index.js

```diff
@@ -307,22 +311,17 @@ class PluginOutgoingSettle extends PluginMiniAccounts {
       }
     }
 
-    if (tag) paymentParams.destination.tag = Number(tag)
-    const tx = await this._api.preparePayment(this._address, paymentParams)
-
-    debug('signing settlement tx. account=', account.getAccount(),
-      'xrp=' + account.getXrpAddressAndTag())
-    const signed = await this._api.sign(tx.txJSON, this._secret)
-    const txHash = signed.id
-    const result = new Promise((resolve, reject) => {
-      this._submitted[txHash] = { resolve, reject }
-    })
-
     debug('submitting settlement tx. account=', account.getAccount(),
       'xrp=' + account.getXrpAddressAndTag())
-    await this._api.submit(signed.signedTransaction)
 
-    await result
+    if (tag) paymentParams.destination.tag = Number(tag)
+    const { resultCode, resultMessage } = await this._txSubmitter('preparePayment', paymentParams)
+
+    if (resultCode !== 'tesSUCCESS') {
```

Comment:
After this PR is [merged](https://github.com/interledgerjs/ilp-plugin-xrp-paychan-shared/pull/10), the tx submitter will validate the result code.

Reply:
So you don't need to check `resultCode` here.